### PR TITLE
DNM: First pass of Couchbase automation

### PR DESCRIPTION
**Pull request by wusui**
Automate Couchbase testing.
The saving of log data is still missing.
The use of a pull-secret yaml with codes in it is probably a bad idea.

Signed-off-by: Warren Usui <warrenusui@localhost.localdomain>

### CODE REVIEW

File: tests/e2e/test_couchbase.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        couchbase.cleanup()
+        ripsaw.cleanup()
+        sc.delete()
+        secret.delete()
+        pool.delete()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+@tier1
```
Comment (RazTamir) [inline tests/e2e/test_couchbase.py]: please change to 'workloads'

File: tests/e2e/test_couchbase.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+class TestCouchbaseWorkload(E2ETest):
+    """
+    Deploy Ycsb workload and run against Couchbase
+    """
+    def test_couchbase_workload_simple(self, ripsaw):
```
Comment (RazTamir) [inline tests/e2e/test_couchbase.py]: please add loggers all over describing each step

File: ocs_ci/ocs/ripsaw.py [hunk 3]
```diff
@@ -76,6 +82,25 @@ def _clone_ripsaw(self):
             log.error('Error during cloning of ripsaw repository')
             raise cf
 
+    def _use_builder_context(self):
```
Comment (vasukulkarni) [inline ocs_ci/ocs/ripsaw.py]: needs docstring

File: ocs_ci/templates/workloads/couchbase/COUCHBASE_server.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  namespace: builder-infra
+spec:
+  infrastructure:
+    name: couchbase
+    args:
+      servers:
+        size: 3
+      storage:
+        use_persistent_storage: True
+        class_name: "rook-ceph-block-extended"
```
Comment (keesturam) [inline ocs_ci/templates/workloads/couchbase/COUCHBASE_server.yaml]: is this storage class name? if so, where is this being created?

File: tests/e2e/test_couchbase.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        log.info("Deploying Ycsb ripsaw operator")
+        ripsaw.apply_crd(
+            'resources/crds/'
+            'ripsaw_v1alpha1_ripsaw_crd.yaml'
+        )
+        # Create ycsb benchmark
+        log.info("Create resource file for ycsb")
+        cb_data = templating.load_yaml_to_dict(constants.YCSB_BENCHMARK_YAML)
+        cb_obj = OCS(**cb_data)
+        cb_obj.create()
```
Comment (keesturam) [inline tests/e2e/test_couchbase.py]: under what conditions will this test fail?

File: ocs_ci/templates/workloads/couchbase/pull_secret.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
...U0RWb1pHeE1VbU5zU0hSZk1IZGZVVFpOUTE5eGRHTXhTemh5TlZkd1NuTlRPSEk0V21WdFZWVlRWRXBQUzNKTGQzTmtZMEZUUm5WVVVFWlpkbloyWXpCdE9VOW5lSE55VjFGdGIzSmxiMncwZDFSRFN6SnhZMmx1ZVZaMmRXdFpTVVZmVTE5Q2VHTnVRemhmU0ZkS2NXVnFRbVYyWDNSVFUwOVFiMWhQUlZoWU9WQmhNbVIxTVhwc1RqUkJNa2huYkhWcVZWcFljbGRqY21kSmFVMHRZVGQ0UzB4V2EzSlBNSEkwTldGU0xUUlZOR1F5V0hkcU4yaHZNVTl3Y1hWMFRqRnNSM28yTTJWb09VbG1TREJRT0hBMlYxQnZXSEJ4WHpKbmIwTkJYemRzZGs5S1prMVFSRVpzVEc5ak1GRktWRTFoUTJ0WWNWb3dVV3hUY0hBeFIwSm9TVnBzUmsxalUxbDRTVlEzZGpaelRIUmxURGhVUkZWTVIyUlFlV3MxUkhwd1gzZHVibFF0VEhoSmNrbzRVVTlHUld0cmRIVnVWamhSVWxGWFFWaHBRUzEzZEU4MVVGWjRNR1ZHUlV0V1pWWlhTbkp1Ums5dFJWcENZM05pYTBRd1QyWTJZV1JXV1dkRVZGVkhTVU5pVmxCUE9YQjRhaTFFVmxGNFF5MXBhbkl0TVdadmRuRnFWRmwwVm1Wak1VWXdTRXg2UVdrM1NrbHdNRkZrZUd0d1ptRnBRamhqVjBJMVVGaFlUMlp4ZDB0RE9XbHJTWEpGTTB0T1gwOTNhMGRYYUdReWQwOWlWMVZ2UmpobVJ6SkRTV2RwY0RSTFQzQXpSRlJvZWxKWVpIQnJWVGhUWlRGSllteHVPVFpGTTBGRlZuVlJXbTlOZG14Nk4zaHZaVk5TYTA1NVUyVlBaVTVQVlVsRGJuVlJabTlQVG0xa016aG1ORk5aVEVORlprZEZSV3BvZUVGSmIwZEtXV0ZSVjNrMWNVZEpSM2s0WlZNNFFuWkVXbEpIVW10Nk5HZG5Xa1pUV2pVeGVtNHhObEozYjFGMlJtaDVTMEZPUldkd2RXaFJMVVk1U2xGcFdESlpVMHhqY0c1SFJFRk1OVk50ZGkxeFZHcDBSR2hPYXpCZlNsOVJMVk40WWxOck5tMWxZUzAxTTBGTmIxRkhkR0pZWTBFMlRtYz0ifX19
```
Comment (keesturam) [inline ocs_ci/templates/workloads/couchbase/pull_secret.yaml]: 2 concerns here.
1) we expose your secret 
2) I doubt if using this secret will be useful when tests are run by other users?

Comment (wusui) [reply ocs_ci/templates/workloads/couchbase/pull_secret.yaml]: I have moved this secret to a data file that will not get checked in.

File: ocs_ci/templates/workloads/ycsb/COUCHBASE_ycsb_loads.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  workload:
+    name: ycsb
+    args:
+      infra: couchbase
+      driver: couchbase2
+      recordcount: 100
+      operationcount: 100
+      requestdistribution: uniform
+      workloads:
+        - workloada
```
[the remark this answers is not in the document]
Comment (wusui) [reply ocs_ci/templates/workloads/ycsb/COUCHBASE_ycsb_loads.yaml]: The ycsb tests consists of several workload tests named workloada, workloadb, workloadc...

File: ocs_ci/templates/workloads/couchbase/COUCHBASE_server.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+apiVersion: builder.cloudbulldozer.io/v1alpha1
+kind: Infra
+metadata:
+  name: couchbase-infra
+  namespace: builder-infra
```
[the remark this answers is not in the document]
Comment (wusui) [reply ocs_ci/templates/workloads/couchbase/COUCHBASE_server.yaml]: I am not sure why.  The cloud-bulldozer/marshall-plan developers named it.

**Comment by vasukulkarni** [comment]
@wusui closing this for now, lets syncup for the reason.

**Pull request closed without merging**